### DISCUSSION TITLE
How to print raw AND styled text in `console.status()`?

### DISCUSSION THREAD

**Discussion by XCanG** [Textualize/rich#3326] [Q&A] [answered]
Here is quick snippet of code that I use to represent the problem:
```py
current_dir, current_file = 0, 0
with console.status("Scanning... Dirs: [bold blue]0[/bold blue] Files: [bold blue]0[/bold blue]") as status:
    for root, dirs, files in source.walk(top_down=True, follow_symlinks=False):
        for name in dirs:
            collect_metadata(root / name, is_dir=True)

        for name in files:
            collect_metadata(root / name)

        current_dir += len(dirs)
        current_file += len(files)
        status.update(f"Scanning... Dirs: [bold blue]{current_dir}[/bold blue] Files: [bold blue]{current_file}[/bold blue] | [yellow]{root}[/yellow]")
```
`source` here is `pathlib.Path` (you can use `os.walk` as alternative), external function `collect_metadata` is not what matter here
The problem comes in `status` string when I update. You can see here is root. Let's assume that `root = "D:\\test\\[Some tag] Some name [end]\\test.txt"`. I use styling to make it with desirable color, however if that name would be printed, it truncate it's value, as if it need to be styled.
![image](https://github.com/Textualize/rich/assets/4011523/70571cc0-47e9-4e9f-9811-d56a7f0324fc)
I couldn't find proper documentation how to print raw styled text in this case or how even escape that. There could be more cases, but this one is more often that I get issues with. There were some options if I use unstyled text, but it is styled in this case, and as it styling it it truncate value (you can see on screenshot).

**Comment by TomJGooding** [comment, marked as answer]
Could you not use `escape`? https://rich.readthedocs.io/en/stable/markup.html#escaping

**Reply by XCanG** [reply]
Oh, I didn't find it. I was remembered that markup used for markdown, so I didn't check that section and look more in text section where I didn't find that.
